packages/panw_cortex_xdr/data_stream/incidents/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,212 @@
+      formats:
+        - UNIX_MS
+      if: ctx.panw_cortex?.xdr?.creation_time != null
+  - date:
+      field: panw_cortex.xdr.resolved_timestamp
+      target_field: panw_cortex.xdr.resolved_timestamp
+      formats:
+        - UNIX_MS
+      if: ctx.panw_cortex?.xdr?.resolved_timestamp != null
+  - date:
+      field: panw_cortex.xdr.modification_time
+      target_field: panw_cortex.xdr.modification_time
+      formats:
+        - UNIX_MS
+      if: ctx.panw_cortex?.xdr?.modification_time != null
+  - set:
+      field: "@timestamp"
+      copy_from: panw_cortex.xdr.creation_time
+      if: ctx.panw_cortex?.xdr?.creation_time != null
+  - set:
+      field: event.severity
+      value: 0
+      if: ctx.panw_cortex?.xdr?.severity == "unknown"
+  - set:
+      field: event.severity
+      value: 1
+      if: ctx.panw_cortex?.xdr?.severity == "informational"
+  - set:
+      field: event.severity
+      value: 2
+      if: ctx.panw_cortex?.xdr?.severity == "low"
+  - set:
+      field: event.severity
+      value: 3
+      if: ctx.panw_cortex?.xdr?.severity == "medium"
+  - set:
+      field: event.severity
+      value: 4
+      if: ctx.panw_cortex?.xdr?.severity == "high"
+  - rename:
+      field: panw_cortex.xdr.incident_id
+      target_field: event.id
+      ignore_missing: true
+  - rename:
+      field: panw_cortex.xdr.description
+      target_field: event.reason
+      ignore_missing: true
+      if: "ctx.panw_cortex?.xdr?.description != null && ctx.panw_cortex.xdr.description instanceof String"
+  - set:
+      field: related.hosts
+      copy_from: panw_cortex.xdr.hosts
+      if: ctx.panw_cortex?.xdr?.hosts != null
+  - set:
+      field: related.user
+      copy_from: panw_cortex.xdr.users
+      if: ctx.panw_cortex?.xdr?.users != null
+  - gsub:
+      field: related.hosts
+      pattern: ":.*"
+      replacement: ""
+      if: ctx.related?.hosts != null
+  - gsub:
+      field: related.user
+      pattern: ".*\\\\"
+      replacement: ""
+      if: ctx.related?.user != null
+  - script:
+      if: ctx.panw_cortex?.xdr?.mitre_techniques_ids_and_names != null
+      lang: painless
+      description: "Extract Mitre Techniques and append it to Threat ECS fields"
+      source: |-
+        void addTechnique(def ctx, def x, def y) {
+          if (ctx?.threat == null) {
+            ctx.threat = new HashMap();
+          }
+          if (ctx?.threat.technique == null) {
+            ctx.threat.technique = new HashMap();
+          }
+          if (ctx?.threat.technique.id == null) {
+            ctx.threat.technique.id = new ArrayList();
+          }
+          if (ctx?.threat.technique.name == null) {
+            ctx.threat.technique.name = new ArrayList();
+          }
+          if (!ctx.threat?.technique?.id.contains(x)) {
+            ctx.threat.technique.id.add(x);
+          }
+          if (!ctx.threat?.technique?.name.contains(y)) {
+            ctx.threat.technique.name.add(y);
+          }
```

```suggestion
          if (!ctx.threat.technique.id.contains(x)) {
            ctx.threat.technique.id.add(x);
          }
          if (!ctx.threat.technique.name.contains(y)) {
            ctx.threat.technique.name.add(y);
          }
```
`threat.technique.{id,name}` are guaranteed to be non-null by the time we reach here.